includes/breadcrumb.inc

```diff
@@ -98,10 +99,15 @@ EOQ;
     $filter_map = function ($filter) {
       return "FILTER($filter)";
     };
-    $query_filters[] = "sameTerm(?collection_predicate, <fedora-rels-ext:isMemberOfCollection>) || sameTerm(?collection_predicate, <fedora-rels-ext:isMemberOf>)";
+    $compound_object_predicate = variable_get('islandora_compound_object_relationship', 'isConstituentOf');
```

Comment:
OK, got an idea. A simple one. Adding a new hook to the islandora api
hook_islandora_get_breadcrumb_query_filters() {
...
}
This way any module can add it's own predicate(s). But this hook should be also implemented in islandora_basic_collection module in order to do it right, so we don't keep this hardcoded.
Would be nice if modules could also suggest a "priority" on it's filters, like in ingest steps. So we can show (i.e, in case of multiple parentship relations) a particular parent and not always the first "dad" this object has.

Sounds good? Too messy?


On Jan 21, 2015, at 7:24 PM, Nick Ruest <notifications@github.com> wrote:

> In includes/breadcrumb.inc:
> 
> > @@ -98,10 +99,15 @@ EOQ;
> >      $filter_map = function ($filter) {
> >        return "FILTER($filter)";
> >      };
> > -    $query_filters[] = "sameTerm(?collection_predicate, <fedora-rels-ext:isMemberOfCollection>) || sameTerm(?collection_predicate, <fedora-rels-ext:isMemberOf>)";
> > +    $compound_object_predicate = variable_get('islandora_compound_object_relationship', 'isConstituentOf');
> @jordandukart just checking back in this so it doesn't get completely lost in the shuffle.
> 
> —
> Reply to this email directly or view it on GitHub.
>